app/vectorstore/opensearch_hybrid.py

```diff
@@ -0,0 +1,215 @@
+        self.vector_store = OpenSearchVectorSearch(
+            index_name=OPENSEARCH_BLOG_DATA_INDEX,
+            embedding_function=self.embeddings,
+            opensearch_url=self.openserach_url,
+            http_auth=self.opensearch_http_auth,
+            use_ssl=True,
+            verify_certs=False,
+            ssl_show_warn=False,
+        )
+
+    def _reciprocal_rank_fusion(
+        self, result_df_list: List[pd.DataFrame], k: int = 60
+    ) -> pd.DataFrame:
+        """Reciprocal Rank Fusion 알고리즘을 사용하여 결과를 결합합니다.
+
+        Args:
+            result_df_list (List[pd.DataFrame]): 검색 결과의 데이터프레임 리스트.
+            k (int): 순위별 가중치 계산을 위한 상수.
+
+        Returns:
+            pd.DataFrame: 결합된 결과의 데이터프레임.
+        """
+        # 결과를 하나의 데이터프레임으로 결합 및 중복 제거
+        df_rrf_result = (
+            pd.concat(result_df_list, axis=0, join="outer")
+            .drop_duplicates()
+            .reset_index(drop=True)
+        )
+
+        # RRF 점수 계산 함수 정의
+        def rrf_score(row):
+            # BM25 순위 관련 RRF 점수 계산
+            rrf_bm25 = (
+                1 / (k + row["BM25_rank"]) if not pd.isna(row["BM25_rank"]) else 0
+            )
+            # 코사인 순위 관련 통한 RRF 점수 계산
+            rrf_cosine = (
+                1 / (k + row["cosine_rank"]) if not pd.isna(row["cosine_rank"]) else 0
+            )
+            return sum([rrf_bm25, rrf_cosine])
+
+        # 각 문서에 대해 RRF 점수를 계산 후, 점수에 따라 정렬
+        df_rrf_result["rrf_score"] = df_rrf_result.apply(rrf_score, axis=1)
+
+        df_rrf_result = df_rrf_result.sort_values(
+            "rrf_score", ascending=False
+        ).reset_index(drop=True)
+
+        return df_rrf_result
+
+    def parse_search_result(self, result: dict, similarity_type: str) -> pd.DataFrame:
+        """검색 결과를 파싱하여 데이터프레임으로 반환합니다.
+
+        Args:
+            result (dict): 검색 결과 데이터.
+            similarity_type (str): 유사성 평가 방법 (예: BM25, cosine).
+
+        Returns:
+            pd.DataFrame: 파싱된 결과의 데이터프레임.
+        """
+        parsed_result = list()
+        # 검색 결과에서 각 문서의 정보 추출
+        for indiv_doc_info in result["hits"]["hits"]:
+            url = indiv_doc_info["_source"]["metadata"]["source"]
+            text = indiv_doc_info["_source"]["text"]
+            title = indiv_doc_info["_source"]["metadata"]["title"]
+            score = indiv_doc_info["_score"]
+            parsed_result.append((url, text, title, score))
+        # 파싱된 결과를 데이터프레임으로 변환 및 랭킹 계산
+        df_parsed = pd.DataFrame(
+            parsed_result, columns=["url", "text", "title", f"{similarity_type}_score"]
+        )
+        df_parsed[f"{similarity_type}_rank"] = df_parsed[
+            f"{similarity_type}_score"
+        ].rank(ascending=False)
+        return df_parsed
+
+    def insert_docs(self, doc_list: List[Document]) -> None:
+        """문서를 인덱스에 삽입합니다.
+
+        Args:
+            doc_list (List[Document]): 삽입할 문서 목록.
+        """
+        # 문서를 벡터 스토어에 삽입
+        self.vector_store.add_documents(doc_list)
```

Comment:
logger를 추가해보면 좋을 것 같네요?! bulk insert라서 요청하는 문서 삽입 사이즈에 따라 소요되는 시간이 달라질텐데요.
logger를 추가해서 진행 상황을 알려주면 좋을 것 같습니다.

지금과 같은 상황이면, 삽입되는 문서 사이즈가 큰경우 콘솔 또는 로그창에서 아무 메시지 없이 멈춰있는 것으로 인지될 수 있어요
```suggestion
    def insert_docs(self, doc_list: List[Document]) -> None:
        """문서를 인덱스에 삽입합니다.

        Args:
            doc_list (List[Document]): 삽입할 문서 목록.
        """
        # 문서를 벡터 스토어에 삽입
        logging.info(f"Inserting {len(doc_list}) document(s)...") 
        self.vector_store.add_documents(doc_list)
        logging.info(f"Done {len(doc_list}) document(s) insertion") 
```

Reply:
[e3f35ae](https://github.com/f-lab-edu/llm-mlops/pull/3/commits/e3f35aec6ba6a02e25147b3270e01938c20e4ea3)에 로깅하는 부분 추가했습니다!
간단한 에러 핸들링 부분도 추가했습니다!

Reply:
문서 삽입에 실패하는 경우가 고려되어야 합니다.  `self.vector_store.add_documents` 함수에서 성공/실패 정보를 반환하고, 삽입 결과를 토대로 실패시 어떠한 행동을 할지 고민해주시기 바래요